services/provider/server/cephblockpool.go

```diff
@@ -83,6 +83,55 @@ func (c *cephBlockPoolManager) SetBootstrapSecretRef(ctx context.Context, blockP
 	return nil
 }
 
+func (c *cephBlockPoolManager) UnSetBootstrapSecretRef(ctx context.Context, blockPoolName, secretName string, cephBlockPool *rookCephv1.CephBlockPool) error {
+
+	// remove the secret ref
+	_, err := ctrl.CreateOrUpdate(ctx, c.client, cephBlockPool, func() error {
+
+		if cephBlockPool.Spec.Mirroring.Peers == nil {
+			cephBlockPool.Spec.Mirroring.Peers = &rookCephv1.MirroringPeerSpec{SecretNames: []string{secretName}}
+		} else {
+			index := slices.IndexFunc(cephBlockPool.Spec.Mirroring.Peers.SecretNames, func(s string) bool {
+				return s == secretName
+			})
+			if index >= 0 {
+				cephBlockPool.Spec.Mirroring.Peers.SecretNames = append(
+					cephBlockPool.Spec.Mirroring.Peers.SecretNames[:index],
+					cephBlockPool.Spec.Mirroring.Peers.SecretNames[index+1:]...)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to unset bootstrap secret ref on CephBlockPool resource with name %q. %v", blockPoolName, err)
+	}
+
+	// delete secret
+	bootstrapSecret := &corev1.Secret{}
+	bootstrapSecret.Name = secretName
+	bootstrapSecret.Namespace = c.namespace
+	err = c.client.Delete(ctx, bootstrapSecret)
+	if err != nil {
+		return fmt.Errorf("failed to delete the bootstrap secret %q. %v", secretName, err)
+	}
+	return nil
+}
+
+func (c *cephBlockPoolManager) DisableBlockPoolMirroring(ctx context.Context, blockPoolName string, cephBlockPool *rookCephv1.CephBlockPool) error {
+
+	_, err := ctrl.CreateOrUpdate(ctx, c.client, cephBlockPool, func() error {
```

Comment:
@rewantsoni could you pls bring this topic in today's team meeting for technical discussion, basically 1. should we use createorupdate on the resources that we didn't create, 2. should be use declarative style while servicing gRPC?

Reply:
As per discussion offline, updated to use Update rather than createOrUpdate for the resources we are not creating